File: Platron/Pages/Helpers/AuthHelper.cs
﻿using System;
using NUnit.Framework;

namespace Platron.Pages.Helpers
{
    public class AuthHelper : BaseHelper
    {
        #region Actions
        public AuthHelper ActionSetLogIn(string login)
        {
            this.ActionFillField(UIAuthPage.FieldLogIn, login);
            return this;
        }

        public AuthHelper ActionSetPassword(string password)
        {
            this.ActionFillField(UIAuthPage.FieldPassword, password);
            return this;
        }

        public AuthHelper ActionClickButtonSubmit()
        {
            UIAuthPage.ButtonSubmit.Click();
            return this;
        }
        #endregion
        #region Checks
        public AuthHelper CheckAlertMessage(string expected)
        {
            Assert.AreEqual(expected, WhenIsClickable(UIAuthPage.AlertMessage).Text.ToString());
            return this;
        }
        #endregion

    }
}

File: Platron/src/Pages/AuthHelper.java
import org.junit.Assert;

public class AuthHelper extends BaseHelper {
    public AuthHelper() {
        super();
    }
    public AuthHelper ActionSetLogIn(String login)
    {
        this.ActionFillField(UIAuthPage.FieldLogIn, login);
        return this;
    }

    public AuthHelper ActionSetPassword(String password)
    {
        this.ActionFillField(UIAuthPage.FieldPassword, password);
        return this;
    }

    public AuthHelper ActionClickButtonSubmit()
    {
        UIAuthPage.ButtonSubmit.click();
        return this;
    }
    public AuthHelper CheckAlertMessage(String expected)
    {
        Assert.assertEquals(expected, WhenIsClickable(UIAuthPage.AlertMessage).getText());
        return this;
    }
}
